Utilities/DataLoading/EdFi.LoadTools/ApiClient/ResourceToInterchangeMapProvider.cs

```diff
@@ -2,69 +2,213 @@
+                    var fileType = element.Name;
 
                     var resources = element.ElementSchemaType.GetUnorderedElementNames().ToList();
 
-                    resources.ForEach(r =>
+                    resources.ForEach(
+                        r =>
+                        {
+                            var resource = r.ToLower();
+
+                            if (!_resourceByFileTypeInfo.TryGetValue(resource, out List<FileTypeInfo> fileTypeInfos))
+                            {
+                                fileTypeInfos = _resourceByFileTypeInfo[resource] = new List<FileTypeInfo>();
+                            }
+
+                            if (!_resourceNameByFileContext.Value.TryGetValue(fileType, out List<FileContext> fileContexts))
+                            {
+                                fileContexts = new List<FileContext>();
+                            }
+
+                            fileTypeInfos.Add(
+                                new FileTypeInfo
+                                {
+                                    FileType = fileType,
+                                    FileContexts = fileContexts
+                                });
+                        });
+                }
+            }
+
+            return _resourceByFileTypeInfo;
+        }
+
+        private Dictionary<string, List<FileContext>> GetFileTypeByFileContext()
+        {
+            var fileTypeByFileContexts = new Dictionary<string, List<FileContext>>();
+
+            foreach (string file in Directory.GetFiles(_dataConfiguration.Folder, "*.xml", SearchOption.AllDirectories))
+            {
+                var contextPrefix = LogContext.BuildContextPrefix(Path.GetFileName(file));
+
+                var sw = new Stopwatch();
+                sw.Start();
+
+                _log.Debug($"{contextPrefix} Start processing");
+
+                FileContext fileContext = CreateFileContext(file, contextPrefix);
+
+                if (!fileContext.IsValid)
+                {
+                    _log.Warn($"{contextPrefix} Failed Validation");
+                    continue;
+                }
+
+                if (fileTypeByFileContexts.TryGetValue(fileContext.FileType, out List<FileContext> fileContexts))
+                {
+                    fileContexts.Add(fileContext);
+                }
+                else
+                {
+                    fileTypeByFileContexts[fileContext.FileType] = new List<FileContext> {fileContext};
+                }
+
+                sw.Stop();
+
+                _log.Debug($"{contextPrefix} Finished processing in {sw.Elapsed.TotalSeconds} seconds.");
+            }
+
+            return fileTypeByFileContexts;
+
+            FileContext CreateFileContext(string fileName, string contextPrefix)
+            {
+                var fileContext = new FileContext
+                {
+                    FileName = fileName,
+                    IsValid = true
```

Comment:
Particularly if you get rid of the try catch, its usually easier to assume invalid, and set it to valid, rather than the opposite.  Should only be one path that ends in valid, but lots of branches that end in invalid.

Reply:
done